### ISSUE TITLE
running my facebook/bart-base for summarization task :  MPS does not support cumsum op with int64 input

### ISSUE THREAD

**Issue by zhannieor** [pytorch/pytorch#141786]
### 🐛 Describe the bug

''' Summarization implementation '''

from domain.interfaces import SummarizationServiceInterface
from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
import torch

class SummarizationService(SummarizationServiceInterface):
    """Summarizes text using a pre-trained model."""

    def __init__(self, model_name: str = "facebook/bart-base", device: str = "mps"):
        """
        Initializes the summarization service.

        Args:
            model_name (str): Name of the pre-trained model to load.
            device (str): Device to run the model on ('cpu' or 'mps').
        """
        self.device = torch.device(device)
        self.tokenizer = AutoTokenizer.from_pretrained(model_name)
        self.model = AutoModelForSeq2SeqLM.from_pretrained(model_name).to(self.device)

    def summarize(self, texts: list[str]) -> str:
        """
        Summarizes a list of texts.

        Args:
            texts (list[str]): A list of text to summarize.

        Returns:
            str: The summarized text.
        """
        combined_text = " ".join(texts)
        inputs = self.tokenizer(
            combined_text, max_length=1024, return_tensors="pt", truncation=True
        ).to(self.device)

        summary_ids = self.model.generate(
            inputs["input_ids"], max_length=150, min_length=40, num_beams=1, early_stopping=True
        )
        return self.tokenizer.decode(summary_ids[0], skip_special_tokens=True)
        
        
        
        The error message is:
        Summarizing Cluster 231 with 153 reviews...
/Users/..../Code/cluster-summ-reviews/venv/lib/python3.9/site-packages/transformers/generation/configuration_utils.py:638: UserWarning: `num_beams` is set to 1. However, `early_stopping` is set to `True` -- this flag is only used in beam-based generation modes. You should set `num_beams>1` or unset `early_stopping`.
  warnings.warn(
INFO:torch.distributed.nn.jit.instantiator:Created a temporary directory at /var/folders/vh/thwrv1sj5cz7rnp9b4lg_wwr0000gn/T/tmpd_c_swrv
INFO:torch.distributed.nn.jit.instantiator:Writing /var/folders/vh/thwrv1sj5cz7rnp9b4lg_wwr0000gn/T/tmpd_c_swrv/_remote_module_non_scriptable.py
/Users/..../Code/cluster-summ-reviews/venv/lib/python3.9/site-packages/transformers/pytorch_utils.py:325: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  test_elements = torch.tensor(test_elements)
ERROR:root:An error occurred: MPS does not support cumsum op with int64 input

### Versions

zsh: command not found: wget

cc @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by hvaara** [comment]
This is likely not a problem with PyTorch itself, as downstream is trying to use `cumsum` with `torch.int64`, which is not supported with the MPS backend. If the error originates in hugging face code I can contrib a fix in their code base.

Can you please provide a minimal reproduction example that can be run top-down to get the same error that you got?

@pytorchbot label "module: mps"

**Comment by hvaara** [comment]
Just rediscovered this issue. My previous answer doesn't really make any sense. int64 support for cumsum on MPS has been supported since PyTorch 2.1.0 and macOS 13.3 (see #95817). Based on the error message `MPS does not support cumsum op with int64 input`, it seems your PyTorch version is from before 2.1.0.

@zhannieor I believe the errors you encountered will be resolved if you update your PyTorch version past 2.1.0, and your macOS version past 13.3. I'll close this issue now, but please feel free to reopen in case you're still seeing errors after updating.

@pytorchbot close
